shared/src/main/scala/mlscript/JSBackend.scala

```diff
@@ -204,8 +177,10 @@ class JSBackend {
     case Bra(_, trm) => translateTerm(trm)
     case Tup(terms) =>
       JSArray(terms map { case (_, term) => translateTerm(term) })
-    case _: Bind | _: Test =>
-      throw CodeGenError(s"cannot generate code for term ${JSBackend.inspectTerm(term)}")
+    case Subs(arr, idx) =>
+      JSMember(translateTerm(arr), translateTerm(idx))
+    case App(_, _) | _: Bind | _: Test =>
+      throw CodeGenError(s"cannot generate code for term ${inspect(term)}")
     case _ => die // FIXME "Exhaustivity analysis reached max recursion depth, not all missing cases are reported."
```

Comment:
Let's try to fix this in this PR. Try with deeper max depth, and if it does not work maybe try to refactor a bit? Such as moving complicated patterns like `case App(App(App(Var("if"), tst), con), alt)` into the `App` case.

Reply:
I extract all cases of `App` into a new method. So exhaustivity check in `translateTerm` becomes easier.

Reply:
Extracting into a different method seems unnecessary, but okay I guess.